sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -79,7 +79,7 @@ trait ExpressionEvalHelper extends ScalaCheckDrivenPropertyChecks with PlanTestB
   private def prepareEvaluation(expression: Expression): Expression = {
     val serializer = new JavaSerializer(new SparkConf()).newInstance()
     val resolver = ResolveTimeZone
-    val expr = resolver.resolveTimeZones(replace(expression))
+    val expr = replace(resolver.resolveTimeZones(expression))
```

Comment:
I think after fixing it here, another PR change can be restored.
https://github.com/apache/spark/pull/48509/files#diff-e306022407781b9b191dd3186e8bb57b9cfee53b6bd5f576257b63bd615fe67cR733
<img width="1030" alt="image" src="https://github.com/user-attachments/assets/de0ab0b2-c08a-407e-a3f1-14234ca9d9c8">